dex/src/main/java/io/crate/data/AsyncCompositeBatchIterator.java

```diff
@@ -117,9 +118,14 @@ public void close() {
 
             // Every "supplyAsync future" completes with the iterator::loadNextBatch() future, so in order to
             // access the loadNextBatch future, the asyncInvocation one should be completed and the result extracted
-            CompletableFuture<CompletableFuture<?>> supplyAsyncFuture =
-                CompletableFuture.supplyAsync(loadNextBatchFuture, executor);
-            asyncInvocationFutures.add(supplyAsyncFuture);
+            try {
+                CompletableFuture<CompletableFuture<?>> supplyAsyncFuture =
+                    CompletableFuture.supplyAsync(loadNextBatchFuture, executor);
+                asyncInvocationFutures.add(supplyAsyncFuture);
+            } catch (RejectedExecutionException e) {
```

Comment:
The executor we currently use on the callsite throws EsRejectedExecutionException - since we don't have that dependency in the dex module I think we'll have to wrap the executor.

Given that we'll have to wrap the Executor anyway, we could also just have a Executor that falls back to direct execution if the delegate rejects it. then the changes here won't be necessary anymore